### DISCUSSION TITLE
Python wrapper for click-to-focus on AM4917MZT4

### DISCUSSION THREAD

**Discussion by fvascellari** [dino-lite/DNX64-Python-API#6] [Q&A]
Good morning,
I've developed a software in Python 3.10 in order to control some linear axes with mounted above an old version of dino-lite (AM4013,T5) to replicate a cheap version of a digital microscope. 

The scope of this implementation is to find the exact position of some point in a grid using the dino-lite as digital microscope and those linear axis to move the grid and catch the exact position, the problem is that the hieght of the grid changes due to multiple factors and, with the previously mentioned version of dino-lite microscope available to us, we are costantly out of focus. For this reason we saw that there is a version (AM4917MZT4) that could adjust the focus with a click in a range of +/- 0.5mm that is what we need.

We have already an offer from your company to purchase the AM4917MZT4 dino-lite model, but before purchase its we want to be sure that there is the possibility to wrap/control the click-to-focus through the python wrapper with the SDK. I've already getted in touch with you customer service to obtain the SDK, but they don't know if the control of click-to-focus is available, colud you please tell me something about?

Thank you in advance

**Comment by charliie-dev** [comment]
Thanks for your interest in our product! 
`DNX64-Python-API` doesn't not provide an direct way to "click-to-focus" via our API; however, our SDK provide control to focus and therefore it's possible to implement through [opencv](https://github.com/jhicks256/OpenCV-Samples/blob/master/autofocus.cpp) . Here is an example for [it](https://pyshine.com/Simple-AutoFocus-Application-in-Python/), you might get some idea from it.

We are considering to provide `click-to-focus` example code, but can not promise an schedule for it.

Feel free to ask more!

**Reply by fvascellari** [reply]
Good morning,

Thank you for your answer, is the microscope capable of adjust its foucs through this example you provide in a range of 0.1mm? 
In case, which is the worst scenario about the date to release a code example for the click to focus?

Thank you

**Reply by charliie-dev** [reply]
To move the microscope len you have to use our API: https://github.com/dino-lite/DNX64-Python-API/blob/31455c7dc051073a7590c8dd6c4af35cdc77525c/DNX64/__init__.py#L51-L53, which is not included in the example I provided.

The example above is to implement autofocus using `opencv` and it's not from our company.
